### PR TITLE
dcos-cli 0.5.1 (new formula)

### PR DESCRIPTION
**Pull request by msabramo**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Formula for [dcos-cli](https://github.com/dcos/dcos-cli)

Cc: @tamarrow

### CODE REVIEW

File: Formula/dcos-cli.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class DcosCli < Formula
+  desc "The command-line for your datacenter!"
+  homepage "https://github.com/dcos/dcos-cli"
+  url "https://downloads.dcos.io/binaries/cli/darwin/x86-64/0.5.1/dcos"
```
Comment (fxcoudert) [inline Formula/dcos-cli.rb]: No binaries, in Homebrew we build from source.

File: Formula/dcos-cli.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  sha256 "9dfcfa2857adf0563765da2a1fc6f006fee3ec8b2ea9174efe7cbfb3822ca42f"
+
+  bottle :unneeded
+
+  depends_on :arch => :x86_64
```
Comment (fxcoudert) [inline Formula/dcos-cli.rb]: And once you build from source, this won't be needed.

File: Formula/dcos-cli.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class DcosCli < Formula
+  desc "The command-line for your datacenter!"
+  homepage "https://github.com/dcos/dcos-cli"
+  url "https://downloads.dcos.io/binaries/cli/darwin/x86-64/0.5.1/dcos"
+  sha256 "9dfcfa2857adf0563765da2a1fc6f006fee3ec8b2ea9174efe7cbfb3822ca42f"
+
+  bottle :unneeded
```
Comment (fxcoudert) [inline Formula/dcos-cli.rb]: Remove that line, after building from source :)

File: Formula/dcos-cli.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on :arch => :x86_64
+
+  def install
+    libexec.install Dir["*"]
+    chmod 0755, libexec/"dcos"
+    bin.install_symlink libexec/"dcos"
+  end
+
+  test do
+    system bin/"dcos", "--version"
```
Comment (fxcoudert) [inline Formula/dcos-cli.rb]: Could we do a stronger test?

File: Formula/dcos-cli.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class DcosCli < Formula
+  desc "The command-line for your datacenter!"
```
Comment (fxcoudert) [inline Formula/dcos-cli.rb]: No `!`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/dcos-cli.rb (hunk 5 above)
Comment (bfontaine) [inline Formula/dcos-cli.rb]: `"Cross-platform command line utility to manage DC/OS clusters"` would be more helpful.

File: Formula/dcos-cli.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "https://github.com/dcos/dcos-cli"
```
Comment (bfontaine) [inline Formula/dcos-cli.rb]: You may want to use https://dcos.io/docs/1.9/cli/ instead of the GitHub repo.

File: Formula/dcos-cli.rb (hunk 1 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Switched to building from source in 44855949c5.

File: Formula/dcos-cli.rb (hunk 3 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Done.

File: Formula/dcos-cli.rb (hunk 2 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: OK.

**Comment by msabramo** [comment]
Switched to building from source in 44855949c5.

File: Formula/dcos-cli.rb (hunk 4 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Stronger test in fa76f64919.

File: Formula/dcos-cli.rb (hunk 6 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Okay, done in 9206e6cd056c9fbba31ff78c5ef08c0b7fbebb90.

File: Formula/dcos-cli.rb (hunk 5 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Okay, done in 9206e6cd056c9fbba31ff78c5ef08c0b7fbebb90.

File: Formula/dcos-cli.rb [hunk 7]
```diff
@@ -0,0 +1,124 @@
+class DcosCli < Formula
+  include Language::Python::Virtualenv
+
+  desc "Cross-platform command-line utility to manage DC/OS clusters"
+  homepage "https://dcos.io/docs/1.9/cli/"
```
Comment (dunn) [inline Formula/dcos-cli.rb]: Is there an unversioned URL for the CLI page (i.e., without the `1.9`)?

Comment (dunn) [inline Formula/dcos-cli.rb]: I guess https://dcos.io/docs/latest/usage/cli/ should work.

File: Formula/dcos-cli.rb [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  url "https://github.com/dcos/dcos-cli/archive/0.5.1.tar.gz"
+  sha256 "5dbf129acc9662ada5eb71024e80f04c44f97b13697dc44a5670eaac148d3222"
+  head "git@github.com:dcos/dcos-cli.git"
```
Comment (dunn) [inline Formula/dcos-cli.rb]: Use `https` for the Git URL too.

File: Formula/dcos-cli.rb [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resources
+    venv.pip_install [buildpath, buildpath/"cli"]
+    bin.install_symlink libexec/"bin/dcos"
+  end
+
+  test do
+    assert_match "dcoscli.version=", shell_output("#{bin}/dcos --version")
+    assert_match "dcos \[options\] \[\<command\>\] \[\<args\>...\]", shell_output("#{bin}/dcos --help")
```
Comment (dunn) [inline Formula/dcos-cli.rb]: Any chance of a test that exercises more of the functionality?

Comment (msabramo) [inline Formula/dcos-cli.rb]: Did you have something in mind? Most of the functionality depends on having a DC/OS cluster, so there's probably not very much that we can do.

File: Formula/dcos-cli.rb (hunk 7 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Cool. See 0476cdbc83.

File: Formula/dcos-cli.rb (hunk 8 above)
Comment (msabramo) [inline Formula/dcos-cli.rb]: Cool. See 0476cdbc83.

**Comment by dunn** [comment]
@fxcoudert I'm happy if you're happy.

**Comment by msabramo** [comment]
@fxcoudert: Any more comments on this one?

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @msabramo for the pull request!

**Pull request closed without merging**